Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/testresult/SubmissionTestResultConsentGivenFragment.kt

```diff
@@ -0,0 +1,133 @@
+
+    private fun navigateToMainScreen() =
+        doNavigate(
+            SubmissionTestResultConsentGivenFragmentDirections
+                .actionSubmissionTestResultConsentGivenFragmentToHomeFragment()
+        )
+
+    private fun buildErrorDialog(exception: CwaWebException): DialogHelper.DialogInstance {
+        return when (exception) {
+            is CwaClientError, is CwaServerError -> DialogHelper.DialogInstance(
+                requireActivity(),
+                R.string.submission_error_dialog_web_generic_error_title,
+                getString(
+                    R.string.submission_error_dialog_web_generic_network_error_body,
+                    exception.statusCode
+                ),
+                R.string.submission_error_dialog_web_generic_error_button_positive,
+                null,
+                true,
+                ::navigateToMainScreen
+            )
+            else -> DialogHelper.DialogInstance(
+                requireActivity(),
+                R.string.submission_error_dialog_web_generic_error_title,
+                R.string.submission_error_dialog_web_generic_error_body,
+                R.string.submission_error_dialog_web_generic_error_button_positive,
+                null,
+                true,
+                ::navigateToMainScreen
+            )
+        }
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        viewModel.uiState.observe2(this) {
+            binding.uiState = it
+            with(binding) {
+                submissionTestResultSection
+                    .setTestResultSection(uiState?.deviceUiState, uiState?.testResultReceivedDate)
+            }
+            it.deviceUiState.withFailure {
+                if (it is CwaWebException) {
+                    DialogHelper.showDialog(buildErrorDialog(it))
+                }
+            }
+        }
+
+        setButtonOnClickListener()
+
+        viewModel.showCancelDialog.observe2(this) {
+            showCancelDialog()
+        }
+
+        viewModel.routeToScreen.observe2(this) {
+            when (it) {
+                is SubmissionNavigationEvents.NavigateToSymptomIntroduction ->
+                    doNavigate(
+                        SubmissionTestResultConsentGivenFragmentDirections
+                            .actionSubmissionTestResultConsentGivenFragmentToSubmissionSymptomIntroductionFragment()
+                    )
+                is SubmissionNavigationEvents.NavigateToMainActivity ->
+                    doNavigate(
+                        SubmissionTestResultConsentGivenFragmentDirections
+                            .actionSubmissionTestResultConsentGivenFragmentToHomeFragment()
+                    )
+            }
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        binding.submissionTestResultContainer.sendAccessibilityEvent(AccessibilityEvent.TYPE_ANNOUNCEMENT)
+    }
```

Comment:
This is an accessibility event. When the screen is resumed it will announce it to a user using TalkBack or a similar assistant - highlighting the important views so that somebody who is perhaps visually impaired knows what is on the screen and how to interact with it.